hathor/nanocontracts/method.py

```diff
@@ -244,25 +277,41 @@ def from_callable(cls, method: Callable) -> Self:
             # XXX: this can (and probably will) be implemented in the future
             if param.default is not EMPTY:
                 raise TypeError('default values are not supported')
+            arg_names.append(param.name)
             args_nc_types.append(make_nc_type_for_arg_type(param.annotation))
 
         return cls(
-            _ArgsNCType(args_nc_types, max_bytes=MAX_BYTES_SERIALIZED_ARG),
-            make_nc_type_for_return_type(method_signature.return_annotation),
+            name=method.__name__,
+            arg_names=arg_names,
+            args_nc_type=_ArgsNCType(args_nc_types, max_bytes=MAX_BYTES_SERIALIZED_ARG),
+            return_nc_type=make_nc_type_for_return_type(method_signature.return_annotation),
         )
 
-    def serialize_args_bytes(self, args: tuple[Any, ...] | list[Any]) -> bytes:
+    def serialize_args_bytes(self, args: tuple[Any, ...] | list[Any], kwargs: dict[str, Any] | None = None) -> bytes:
         """ Shortcut to serialize args directly to a bytes instead of using a serializer.
         """
-        return _serialize_map_exception(self.args, args)
+        if len(args) > len(self.arg_names):
+            raise NCFail('too many arguments')
+
+        merged: dict[str, Any] = {}
+        for index, arg in enumerate(args):
+            name = self.arg_names[index]
+            merged[name] = arg
+
+        kwargs = kwargs or {}
+        for name, arg in kwargs.items():
+            if name not in self.arg_names:
+                raise NCFail(f"{self.name}() got an unexpected keyword argument '{name}'")
+            if name in merged:
+                raise NCFail(f"{self.name}() got multiple values for argument '{name}'")
+            merged[name] = arg
+
+        return _serialize_map_exception(self.args, tuple(merged.values()))
```

Comment:
I guess the safest here would be the following. The main different is that it does not depend on dicts keeping the insertion order (which is something recent in Python).

```suggestion
        return _serialize_map_exception(self.args, tuple(merged[name] for name in self.arg_names))
```

Reply:
Alternatively we could explicitly use an `OrderedDict`